### PR TITLE
TONS of QoL improvements

### PR DESCRIPTION
**Pull request by TacoTheDank**
I suppose every commit speaks for itself, because making the list of changes in each commit like I usually do in other PRs would be too long and confusing here (17 commits lol).

I've built and tested APKs the whole way through, and they work great :)

Please leave comments if you have any questions, thanks!

Here's an APK for testing (it has a slightly different package name so you can test the current version and this build side-by-side: [app-release.zip](https://github.com/fenimore/DemocracyDroid/files/6225000/app-release.zip)

### CODE REVIEW

**Comment by fenimore** [comment]
Hi! Thanks so much I'll take a look soon :relaxed:

File: app/src/main/java/com/workingagenda/democracydroid/MediaService.java [hunk 1]
```diff
@@ -88,16 +58,18 @@ public IBinder onBind(Intent intent) {
...
         LoadControl loadControl = new DefaultLoadControl();
 
         // Create Player
         player = new SimpleExoPlayer.Builder(getApplicationContext())
-                .setLoadControl(loadControl).setTrackSelector(trackSelector).build();
+                .setLoadControl(loadControl)
+                .setTrackSelector(trackSelector)
+                .build();
 
-        player.setPlayWhenReady(true);
```
Comment (fenimore) [inline app/src/main/java/com/workingagenda/democracydroid/MediaService.java:100]: How come you changed this to `play`?

File: app/src/main/java/com/workingagenda/democracydroid/SettingsActivity.java [hunk 2]
```diff
@@ -1,168 +1,40 @@
...
-        }
-
-        ListPreference open = (ListPreference) findPreference("open_preference");
-        if(open.getValue() == null){
-            open.setValueIndex(0);
-        }
-        String versionName = BuildConfig.VERSION_NAME;
-        Preference versionPref = findPreference("pref_static_field_key0");
-        Preference newPref = findPreference("whats_new");
-        versionPref.setSummary("Democracy Droid! " + versionName);
```
Comment (fenimore) [inline app/src/main/java/com/workingagenda/democracydroid/SettingsActivity.java:59]: Why did you remove all the settings activity settings?

Comment (fenimore) [reply app/src/main/java/com/workingagenda/democracydroid/SettingsActivity.java:59]: Oh nevermind I see you translated it into a fragment :+1:

File: app/src/main/res/layout/activity_about.xml [hunk 3]
```diff
@@ -1,97 +1,113 @@
...
-                style="?android:attr/listSeparatorTextViewStyle"/>
+                android:text="@string/about_dev_header" />
+
             <TextView
-                android:id="@+id/about_4"
+                android:id="@+id/about_info"
                 android:layout_width="wrap_content"
                 android:layout_height="wrap_content"
-                android:textSize="16sp"
+                android:autoLink="web"
```
Comment (fenimore) [inline app/src/main/res/layout/activity_about.xml:90]: nice! I did not know about this!

File: gradlew [hunk 4]
```diff
@@ -1,25 +1,59 @@
-#!/usr/bin/env bash
```
Comment (fenimore) [inline gradlew:1]: Hmm, I don't really understand these changes, or the gradle binary files that have been checked into git, could you give a little explanation for what these changes are about? Thank you!

**Comment by fenimore** [comment]
Hi @TacoTheDank ! Thank you very much for these updates -- 

I see there is some `binding` pattern you're using, which I more or less understand

As I left in a comment, could you speak to the gradle binary files and script updates you provided?

Also I figure you used some automated tooling for some of this refactoring? If so, could you please provide a little explanation as to what tools you used (for my own edification :relaxed: )

Thanks again!

File: app/src/main/java/com/workingagenda/democracydroid/MediaService.java (hunk 1 above)
Comment (TacoTheDank) [reply app/src/main/java/com/workingagenda/democracydroid/MediaService.java:100]: https://github.com/google/ExoPlayer/blob/release-v2/RELEASENOTES.md#2120-2020-09-11

> Add play and pause convenience methods. They are equivalent to setPlayWhenReady(true) and setPlayWhenReady(false) respectively.

File: gradlew (hunk 4 above)
Comment (TacoTheDank) [reply gradlew:1]: I updated the gradle wrapper using git bash, and that automatically updates the gradle binary files to those versions.

**Comment by TacoTheDank** [comment]
@fenimore 

> As I left in a comment, could you speak to the gradle binary files and script updates you provided?

Yes, take a look 👍 

> Also I figure you used some automated tooling for some of this refactoring? If so, could you please provide a little explanation as to what tools you used (for my own edification ☺️ )

Sure!

`Reformat codestyle to an .editorconfig` utilized the code reformatting tool (shortcut is Ctrl+Alt+L).

Some of `Migrate to AppCompat where useful` used the "Migrate to AppCompat" option found in AS's Refactor tab.

`Convert icons to vector drawables` utilized AS's vector asset generation tool.

`Automatic cleanup by Studio` was done using Ctrl+Alt+L like before, but checking the "cleanup code" option as well.

`Lint cleanup` is one where I really just went through and manually corrected a bunch of lints, but there's linting tools in AS's Analyze tab (the first three options) that can inspect the project for lints, as well as correct them.

Everything else was pretty much done manually.

**Comment by fenimore** [comment]
Amazing thank you again! I'll try to keep these clean code tools in mind :relaxed:

**Comment by TacoTheDank** [comment]
@fenimore I noticed you bumped the version name and code. Did you upload a new version to the play store?

**Comment by fenimore** [comment]
Hi! Yes, so I've updated the release. This will automatically update f droid. 

I haven't added this app to the Play Store yet actually, I was thinking I'd wait a little while before doing so and see how Fdroid people like it

**Comment by fenimore** [comment]
Oh, actually @TacoTheDank I was confused earlier, I thought this was for a different application! This Is very much on the Play Store! and I haven't released it yet I was testing it for a little while (I thought maybe a week or so), before releasing to the Playstore
